lean4/src/putnam_1985_b1.lean

```diff
@@ -13,9 +13,11 @@ p(x) = (x-m_1)(x-m_2)(x-m_3)(x-m_4)(x-m_5)
 has exactly $k$ nonzero coefficients. Find, with proof, a set of integers $m_1, m_2, m_3, m_4, m_5$ for which this minimum $k$ is achieved.
 -/
 theorem putnam_1985_b1
-(p : (Fin 5 → ℤ) → (Polynomial ℝ))
-(hp : p = fun m ↦ ∏ i : Fin 5, ((X : Polynomial ℝ) - m i))
-(numnzcoeff : Polynomial ℝ → ℕ)
-(hnumnzcoeff : numnzcoeff = fun p ↦ {j ∈ Finset.range (p.natDegree + 1) | coeff p j ≠ 0}.ncard)
-: (Injective putnam_1985_b1_solution ∧ ∀ m : Fin 5 → ℤ, Injective m → numnzcoeff (p putnam_1985_b1_solution) ≤ numnzcoeff (p m)) :=
-sorry
+    (p : (Fin 5 → ℤ) → (Polynomial ℝ))
+    (hp : ∀ m, p m = ∏ i, ((X : Polynomial ℝ) - m i))
+    (numnzcoeff : Polynomial ℝ → ℕ)
+    (hnumnzcoeff : ∀ p, numnzcoeff p =
+      Set.ncard {j ∈ Finset.range (p.natDegree + 1) | coeff p j ≠ 0}) :
```

Comment:
I guess these could switch to using `.card` if the set comprehension is now a `Finset`?

Reply:
Yes, I was trying to be ultraconservative and not change the meaning but if George prefers I could make such changes.

Reply:
Yeah I agree it makes sense to switch to .card in this setting. We can also make those changes if it's easier.

Reply:
OK I've changed these to use `Finset.card`.